server/rest/src/main/java/org/infinispan/rest/resources/MetricsResource.java

```diff
@@ -67,41 +71,50 @@ public Invocations getInvocations() {
    }
 
    private CompletionStage<RestResponse> metrics(RestRequest restRequest) {
-      try {
-         List<String> accept = restRequest.headers(HttpHeaderNames.ACCEPT.toString());
-
-         // provide defaults for missing ACCEPT header (based on http method)
-         if (restRequest.method() == GET) {
-            if (accept.isEmpty()) {
-               // default to OpenMetrics (Prometheus) if nothing specified
-               accept = Collections.singletonList(TEXT_PLAIN_TYPE);
-            } else {
-               // to handle OpenMetrics we need to swap it to "text/plain" so smallrye can recognize it
-               accept = accept.stream()
-                              .map(h -> h.startsWith(APPLICATION_OPENMETRICS_TYPE) ? TEXT_PLAIN_TYPE : h)
-                              .collect(Collectors.toList());
-            }
-         } else if (restRequest.method() == OPTIONS) {
-            if (accept.isEmpty()) {
-               accept = Collections.singletonList(APPLICATION_JSON_TYPE);
-            }
-         }
+      List<String> accept = responseMediaTypes(restRequest);
 
+      CompletableFuture<RestResponse> cf = new CompletableFuture<>();
+      blockingExecutor.execute(() -> {
```

Comment:
I would recommend using CompletableFuture.supplyAsync instead so we don't have to do the CompletableFuture complete calls ourselves.